shared/redux/actions/btcmultisig.js

```diff
@@ -0,0 +1,260 @@
+      reducers.user.setBalance({ name: 'btcMultisigData', amount: balance, unconfirmedBalance })
+      return balance
+    })
+    .catch((e) => {
+      reducers.user.setBalanceError({ name: 'btcMultisigData' })
+    })
+}
+
+const fetchBalance = (address) =>
+  request.get(`${api.getApiServer('bitpay')}/addr/${address}`)
+    .then(({ balance }) => balance)
+
+const fetchTx = (hash) =>
+  request.get(`${api.getApiServer('bitpay')}/tx/${hash}`)
+    .then(({ fees, ...rest }) => ({
+      fees: BigNumber(fees).multipliedBy(1e8),
+      ...rest,
+    }))
+
+const fetchTxInfo = (hash) =>
+  fetchTx(hash)
+    .then(({ vin, ...rest }) => ({
+      senderAddress: vin ? vin[0].addr : null,
+      ...rest,
+    }))
+
+const getTransaction = () =>
+  new Promise((resolve) => {
+    const { user: { btcData: { address } } } = getState()
+
+    const url = `${api.getApiServer('bitpay')}/txs/?address=${address}`
+
+    return request.get(url)
+      .then((res) => {
+        const transactions = res.txs.map((item) => {
+          const direction = item.vin[0].addr !== address ? 'in' : 'out'
+          const isSelf = direction === 'out'
+            && item.vout.filter((item) =>
+              item.scriptPubKey.addresses[0] === address
+            ).length === item.vout.length
+
+          return ({
+            type: 'btc',
+            hash: item.txid,
+            confirmations: item.confirmations,
+            value: isSelf
+              ? item.fees
+              : item.vout.filter((item) => {
+                const currentAddress = item.scriptPubKey.addresses[0]
+
+                return direction === 'in'
+                  ? (currentAddress === address)
+                  : (currentAddress !== address)
+              })[0].value,
+            date: item.time * 1000,
+            direction: isSelf ? 'self' : direction,
+          })
+        })
+        resolve(transactions)
+      })
+      .catch(() => {
+        resolve([])
+      })
+  })
+
+const send = async ({ from, to, amount, feeValue, speed } = {}) => {
+  feeValue = feeValue || await btc.estimateFeeValue({ inSatoshis: true, speed })
+  const { user: { btcMultisigData: { address, privateKey, publicKeys, publicKey } } } = getState()
+
+  const unspents      = await fetchUnspents(from)
+
+  const fundValue     = new BigNumber(String(amount)).multipliedBy(1e8).integerValue().toNumber()
+  const totalUnspent  = unspents.reduce((summ, { satoshis }) => summ + satoshis, 0)
+  const skipValue     = totalUnspent - fundValue - feeValue
+
+  const p2ms = bitcoin.payments.p2ms({
+    m: 2,
+    pubkeys: publicKeys,
+    network: btc.network,
+  })
+  const p2sh = bitcoin.payments.p2sh({ redeem: p2ms, network: btc.network })
+  let txb1 = new bitcoin.TransactionBuilder(btc.network)
+  unspents.forEach(({ txid, vout }) => txb1.addInput(txid, vout))
+  txb1.addOutput(to, fundValue)
+  txb1.__INPUTS.forEach((input, index) => {
```

Comment:
либа биткоин джс говорит, что надо делать цикл не по инпутам, а по тому массиву, который ты туда кладешь. Я не помню, почему они это гвоорят, но можно почитать. Типа так должно быть:
```suggestion
unspents.forEach((utxo, index) => {
```

Reply:
От этого ничего не должно измениться, ведь на `__INPUTS` мы не воздействуем, а индексы мы раньше брали именно из инпутов